src/hotspot/share/compiler/compilerDirectives.hpp

```diff
@@ -185,6 +185,39 @@ class ControlIntrinsicIter {
   ControlIntrinsicIter& operator++();
 };
 
+class ControlIntrinsicValidator {
+ private:
+  bool _valid;
+  char* _bad;
+
+ public:
+  ControlIntrinsicValidator(ccstrlist option, bool disabled_all) : _valid(true), _bad(nullptr) {
+    for (ControlIntrinsicIter iter(option, disabled_all); *iter != NULL && _valid; ++iter) {
+      if (vmIntrinsics::_none == vmIntrinsics::find_id(*iter)) {
+        const size_t len = MIN2<size_t>(strlen(*iter), 63) + 1;  // cap len to a value we know is enough for all intrinsic names
+        _bad = NEW_C_HEAP_ARRAY(char, len, mtCompiler);
+        // strncpy always write len characters.if the source string is shorter, the function fills the remaining bytes with NULs.
```

Comment:
Some typos, should be: `// strncpy always writes len characters. If the source string is shorter, the function fills the remaining bytes with NULLs.`